Support displaying team seeds in tournament client

frenzibyte opened the pull request:
As per request from the OWC team. Preview:

https://github.com/ppy/osu/assets/22781491/28b95f89-ce3b-4db2-8d11-cf148824eca3

Also, I've adjusted the padding of the tournament sprite texts to be horizontally equal, as previously it was padded 10px more from the right, which looked completely off on a text with numbers only:

![CleanShot 2023-10-28 at 08 52 33](https://github.com/ppy/osu/assets/22781491/32ee85b8-9015-4687-a2a5-b35da03dbc72)

---

EDIT: This also shows up in the "team intro" screen now:

![CleanShot 2023-10-28 at 09 28 48](https://github.com/ppy/osu/assets/22781491/e720549d-97e4-4b6d-9947-22b2827d0db5)

frenzibyte committed:
Support displaying team seed in `TeamDisplay`

frenzibyte committed:
Add setting to configure team seed display

frenzibyte committed:
Adjust horizontal padding in tournament sprite text

frenzibyte committed:
Move team seed setting from gameplay screen

frenzibyte committed:
Display team seeds in map pool screen as well

frenzibyte committed:
Refactor logic slightly to display team seed everywhere

This change makes the team seed display in "team intro" screen as well.

frenzibyte committed:
Fix failing tests

bdach commented:
> I've adjusted the padding of the tournament sprite texts to be horizontally equal, as previously it was padded 10px more from the right, which looked completely off on a text with numbers only:
> 
> ![CleanShot 2023-10-28 at 08 52 33](https://user-images.githubusercontent.com/22781491/278794067-32ee85b8-9015-4687-a2a5-b35da03dbc72.png)

I'm pretty sure this was a 200% intentional design decision, and as such it should be probably reverted. For what it's worth, I don't think the attached screenshot even looks that bad. If anything I am bothered more by the new horizontally symmetric spacing, since it begins to draw attention to how _low_ vertically the text is on these, and _that_ begins to look wrong.

I'm also not sure that the seed display should be that large. It's not as important as to use the same size of text as the team name.

If I were to propose anything:

![1698667639](https://github.com/ppy/osu/assets/20418176/e08b3fed-d437-46fd-8332-dbbd4e0aa00d)

but I'm not a designer. Maybe @arflyte can offer something up.

frenzibyte commented:
> > I've adjusted the padding of the tournament sprite texts to be horizontally equal, as previously it was padded 10px more from the right, which looked completely off on a text with numbers only:
> > ![CleanShot 2023-10-28 at 08 52 33](https://user-images.githubusercontent.com/22781491/278794067-32ee85b8-9015-4687-a2a5-b35da03dbc72.png)
> 
> I'm pretty sure this was a 200% intentional design decision, and as such it should be probably reverted. For what it's worth, I don't think the attached screenshot even looks that bad. If anything I am bothered more by the new horizontally symmetric spacing, since it begins to draw attention to how _low_ vertically the text is on these, and _that_ begins to look wrong.
> 
> I'm also not sure that the seed display should be that large. It's not as important as to use the same size of text as the team name.
> 
> If I were to propose anything:
> 
> ![1698667639](https://user-images.githubusercontent.com/20418176/279049615-e08b3fed-d437-46fd-8332-dbbd4e0aa00d.png)
> 
> but I'm not a designer. Maybe @arflyte can offer something up.



> > I've adjusted the padding of the tournament sprite texts to be horizontally equal, as previously it was padded 10px more from the right, which looked completely off on a text with numbers only:
> > ![CleanShot 2023-10-28 at 08 52 33](https://user-images.githubusercontent.com/22781491/278794067-32ee85b8-9015-4687-a2a5-b35da03dbc72.png)
> 
> I'm pretty sure this was a 200% intentional design decision, and as such it should be probably reverted. For what it's worth, I don't think the attached screenshot even looks that bad. If anything I am bothered more by the new horizontally symmetric spacing, since it begins to draw attention to how _low_ vertically the text is on these, and _that_ begins to look wrong.

Sure. Definitely my bad for combining such major change with a small PR like this, but I'll revert anyway.

peppy commented:
I think the smaller version proposal looks better, let's go with that.

peppy committed:
Move team seed to below team name

peppy committed:
Adjust `DrawableTeamTitleWithHeader` to match new layout

peppy committed:
Merge branch 'master' into tournament-display-team-seed
